### ISSUE TITLE
How does this script get the name of the Wifi network you are connected to?

### ISSUE THREAD

**Issue by FlexMcMurphy** [edigiacomo/android-rsync-backup#1]
Hello,

I would like to use your script in my phone.

Can you please explain how this line:
`if [[ "$ssid" != "termux-wifi-connectioninfo | jq -r '.ssid')" ]]`
... gets the name of the Wifi network the phone is connected to?

What does this string mean or contain --> `termux-wifi-connectioninfo` ?
Am I supposed to have a .ssid file in the same folder as the script?

Cheers,

Flex

**Comment by edigiacomo** [comment]
Hi! First of all, there was a typo in that line that I have just fixed in commit 947eda5.

`termux-wifi-connectioninfo` is a command that prints to stdout a JSON containing details about the WiFi network the phone is connected to. One of the JSON keys is `"ssid"` (the name of the WiFi network) and the command `jq -r '.ssid'` prints only that key.

**Comment by FlexMcMurphy** [comment]
Very cool and thanks for this useful script!

I will modify it to allow connection from two different networks so I can sync when connected to my LAN or when outside the LAN and connected to it via a VPN over another specified Wifi network.

I found the command `termux-wifi-connectioninfo` returns: `"ssid_hidden": true` and `"ssid": "<unknown ssid>"` unless you enable Termux to have access to your phones location:

Phone Settings > Privacy > Permission Manager > Location.

I got help on that [from here](https://github.com/termux/termux-api/issues/304).

Cheers,

Flex
